miso-web/src/main/webapp/scripts/list_partition.js

```diff
@@ -0,0 +1,305 @@
+                }), 'Outstanding Orders (All)', assign, assignDialog),
+            assignFromRest('/miso/rest/pool/picker/readytorun?' + jQuery
+                .param({
+                  platform : platformType.name,
+                  readyToRun : true
+                }), 'Ready to Run', assign, assignDialog),
+            assignFromRest('/miso/rest/pool/picker/recent?' + jQuery.param({
+              platform : platformType.name
+            }), 'Recently Modified', assign, assignDialog), ]
+            .filter(function(x) {
+              return x;
+            });
+        assignDialog();
+      }
+    } ];
+    if (config.runId) {
+      actions.push({
+        name : "Set QC",
+        action : function(partitions) {
+          var setQc = function(id, notes) {
+            Utils.ajaxWithDialog('Setting QC', 'POST',
+                '/miso/rest/run/' + config.runId + '/qc', {
+                  "partitionIds" : partitions.map(Utils.array.getId),
+                  "qcTypeId" : id,
+                  "notes" : notes
+                }, Utils.page.pageReload);
+          };
+          
+          Utils.showWizardDialog("Set QC", Constants.partitionQcTypes
+              .map(function(qcType) {
+                return {
+                  name : qcType.description,
+                  handler : function() {
+                    if (qcType.noteRequired) {
+                      Utils.showDialog(qcType.description + " Notes", "Set",
+                          [ {
+                            type : "text",
+                            label : "Notes",
+                            property : "notes"
+                          } ], function(results) {
+                            setQc(qcType.id, results.notes);
+                          });
+                    } else {
+                      setQc(qcType.id, null);
+                    }
+                  }
+                };
+              }));
+        }
+      });
+    }
+    return actions;
+  },
+  createStaticActions : function(config, projectId) {
+    return [];
+  },
+  createColumns : function(config, projectId) {
+    return [
+        {
+          "sTitle" : "Container",
+          "mData" : "containerName",
+          "include" : config.showContainer,
+          "iSortPriority" : 0
+        },
+        {
+          "sTitle" : "Number",
+          "mData" : "partitionNumber",
+          "include" : true,
+          "iSortPriority" : 1,
+          "bSortDirection" : true
+        },
+        {
+          "sTitle" : "Pool",
+          "mData" : "pool",
+          "include" : true,
+          "iSortPriority" : 0,
+          "mRender" : function(data, type, full) {
+            if (!data) {
+              if (type === 'display') {
+                return "(None)";
+              } else {
+                return "";
+              }
+            }
+            var prettyName = data.name + " (" + data.alias + ")";
+            if (type === 'display') {
+              return data
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Change this condition so that it does not always evaluate to "true"; some subsequent code is never executed. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=javascript%3AS2583)